src/runtime/security/server_negotiation.h

```diff
@@ -21,13 +21,19 @@
 
 namespace dsn {
 namespace security {
+extern const std::set<std::string> supported_mechanisms;
 
 class server_negotiation : public negotiation
```

Comment:
Why is `rpc_session` holding a "server_negotiation" with no data? What gonna be inside `server_negotiation`?

Reply:
`server_negotiation` have some members. I will add them in later pull requests.

Reply:
Can you list them? Just check if the overall design is good.

Reply:
`std::string _selected_mechanism` is the mechanism selected by negotiation
`std::unique_ptr<sasl_conn_t, sasl_deleter> _sasl_conn` is the sasl connection context

Reply:
Ok. You gave a quite bad name. So the class `negotiation` is actually "sasl_context". I think the latter is more descriptive. I don't have to check the code to know the overall design.

“server_negatiation” is "server_sasl_context".

Reply:
No, it's not only `sasl_context`. The class `negotiation` should also do negotiation work to select mechansim.